lib/reek/spec/should_reek_of.rb

```diff
@@ -8,6 +8,7 @@ module Spec
     # An rspec matcher that matches when the +actual+ has the specified
     # code smell.
     #
+    # :reek:InstanceVariableAssumption
```

Comment:
Is there a specific reason for `@matching_smell_types = nil` not live in `initialize`?